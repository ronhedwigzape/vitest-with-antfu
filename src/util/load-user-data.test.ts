import { describe, test, expect } from 'vitest'
import { loadUserData } from "./load-user-data.ts";

describe('loadUserDetails', () => {
    test('load user data as expected', async () => {
        const user = await loadUserData('antfu7');

        expect(user).toMatchSnapshot()
    })

    test('sets coolness level appropriately', async () => {
        const jason = await loadUserData('jilengstorf')
        const anthony = await loadUserData('antfu7')

        expect(jason.coolness).toBe(-1);
        expect(anthony.coolness).toBe(100)
    })

    // test.fails('throws an error for nonexistent users', async () => {
    //     const notReal = await loadUserData('fakeUser');
    //
    // })

    test('throws an error for nonexistent users', async () => {
        expect(async () => await loadUserData('fakeUser')).rejects.toThrowError(
            'no user found');
    })

})
