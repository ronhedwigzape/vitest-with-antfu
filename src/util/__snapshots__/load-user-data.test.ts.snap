// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`loadUserDetails > load user data as expected 1`] = `
{
  "coolness": 100,
  "name": "Anthony",
  "projects": [
    "vitest",
    "vite",
  ],
  "username": "antfu7",
}
`;
